### PR TITLE
feat(dev_env): Dev env bootstrap support for two Python versions

### LINKED ISSUE #28154: Support dual Python dev env
3.8.10 for Apple's M1 and 3.6.13 everywhere else

### PR DESCRIPTION
**Pull request by armenzg**
This change adds supports for bootstrapping two different Python versions.
Python 3.8.10 will be used for Apple M1 since we're targetting Python 3.8 for Sentry's
next Python version and it is the first one that supports Apple's M1 chipset.

Fixes: #28154

### CODE REVIEW

File: scripts/lib.sh
Comment (joshuarli) [reply scripts/lib.sh:54]: I think it's better to be strict about it so we can keep people's devenvs as close as possible to production. Also, 3.8.11 is latest.

Comment (joshuarli) [reply scripts/lib.sh:54]: Also, if we could hold off on merging this, that would be ideal since we're not yet ready yet for 3.8. Could separate some of your other refactors into separate PRs.

File: .envrc
Comment (billyvg) [reply .envrc:7]: What about?

```suggestion
SENTRY_ROOT="$(
```

File: .envrc
Comment (billyvg) [reply .envrc:169]: Yeah if we are doing this only for `bootstrap-py3-venv`, it makes more sense to call it from there.

File: .envrc
Comment (billyvg) [inline .envrc:226]: I like having a verb in here to better show what it's doing e.g.

```suggestion
    show_commands_info
```

File: .github/workflows/api-docs-test.yml
Comment (billyvg) [inline .github/workflows/api-docs-test.yml:33]: Debugging?

File: scripts/ensure-venv.sh
Comment (billyvg) [inline scripts/ensure-venv.sh:25]:
```suggestion
    # The developer is inside a virtualenv *and* has set a SENTRY_PYTHON_VERSION
```

File: scripts/lib.sh
Comment (billyvg) [inline scripts/lib.sh:33]: I think if the return value is a boolean, we should rename to be `is-<object>`

File: scripts/lib.sh
Comment (billyvg) [inline scripts/lib.sh:49]: And here since we're returning the current pyenv version, I would use `get` instead of `query`

File: scripts/lib.sh
Comment (billyvg) [inline scripts/lib.sh:51]:
```suggestion
# XXX: Fix this check to be more lenient
```

File: scripts/lib.sh
Comment (billyvg) [reply scripts/lib.sh:33]: Seems like we have a mix of using `_` and `-` as separators. (e.g. in `.envrc`)

File: scripts/lib.sh
Comment (billyvg) [inline scripts/lib.sh:52]: I think using the singular `version` is more correct here?

File: .envrc
Comment (armenzg) [reply .envrc:169]: Sounds good.

File: .github/workflows/api-docs-test.yml
Comment (armenzg) [reply .github/workflows/api-docs-test.yml:33]: Yes :)

File: requirements-base.txt
Comment (armenzg) [inline requirements-base.txt:24]: See https://github.com/googleapis/python-crc32c/issues/83

These google dependencies might end up moving into the dev requirements (#28305).

File: scripts/lib.sh
Comment (billyvg) [reply scripts/lib.sh:33]: Hmm I'd probably use it when generically interacting with a service or something along those lines. e.g. `query_google(search_term)`

File: .github/workflows/api-docs-test.yml
Comment (billyvg) [reply .github/workflows/api-docs-test.yml:33]: Should it be removed before merging?

File: .github/workflows/python-deps.yml
Comment (billyvg) [reply .github/workflows/python-deps.yml:6]: What is doing the auto formatting?

File: .github/actions/setup-python/action.yml
Comment (armenzg) [inline .github/actions/setup-python/action.yml:31]: @billyvg you won the race to merge :)

File: scripts/bootstrap-py3-venv
Comment (armenzg) [inline scripts/bootstrap-py3-venv:34]: I need to set the variable so `pyenv` will know what Python version to change to.
I've tested this on both my machines.

File: .github/actions/setup-python/action.yml
Comment (billyvg) [reply .github/actions/setup-python/action.yml:31]: aha sorry!

File: .github/workflows/python-deps.yml
Comment (armenzg) [reply .github/workflows/python-deps.yml:6]: The Prettier VS Code extension (we recommend it).

I spend some time looking into it. It does not always execute to completion, thus, it does not always auto-format.
image

**Comment by armenzg** [comment]
@billyvg @joshuarli this is ready for another review. Thanks!

You can ignore the dev env workflow failure. I will fix that in #28265

File: .github/workflows/python-deps.yml
Comment (billyvg) [reply .github/workflows/python-deps.yml:6]: Ah we should make this run on precommit

**Review by billyvg — APPROVED** [review]

File: .github/workflows/python-deps.yml
Comment (armenzg) [reply .github/workflows/python-deps.yml:6]: Added to backlog

**Comment by armenzg** [comment]
I will follow up on any other comments on my next PR.

**Pull request merged**
